bigchaindb/consensus.py

```diff
@@ -139,12 +139,21 @@ def validate_transaction(bigchain, transaction):
                 raise exceptions.OperationError(
                     'Only federation nodes can use the operation `CREATE`')
 
+            # validate digital asset
+            # A create transaction will have only one output
+            asset = transaction['transaction']['asset']
+            assets.validate_asset_creation(asset['data'], asset['divisible'], asset['updatable'],
+                                           asset['refillable'], transaction['transaction']['conditions'][0]['amount'])
+            # TODO: Validate uuid?
```

Comment:
The change you proposed will add a bit more code to create_tx but it will simplify validate_transaction which I think is a good tradeoff since we want validate_transaction to be as fast as possible